### PR TITLE
Enhance compatibility with different Hive shells

### PR DESCRIPTION
**Pull request by teabot**
Both the `hive` and `beeline` shells exhibit subtly different behaviours with regard to full line comments. The behaviour can also differ between the file (`-f` option) mode or interactive mode of the same shell. This pull request introduces a `CompatibilityMode` so that the user can select which behaviours they expect in tests when using HiveRunner. The mode defaults to that provided by the `hive` command.

Such differences should become irrelevant once [HIVE-8296](https://issues.apache.org/jira/browse/HIVE-8396) is resolved and released, although the reality is that it will take some time to filter down into general deployment. Therefore this PR aims to provide a more accurate Hive test environment for what is currently available.

Related to issue https://github.com/klarna/HiveRunner/issues/15

### CODE REVIEW

File: src/main/java/com/klarna/hiverunner/CompatibilityMode.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2013 Klarna AB
```
Comment (massdosage) [inline src/main/java/com/klarna/hiverunner/CompatibilityMode.java]: Should be 2015 (applies to other new licenses below too)

**Comment by PelleUllberg** [comment]
Hi, 

Could we rather have this configured via the com.klarna.hiverunner.config.HiveRunnerConfig? I'm thinking that this is something you'd want to set for a whole test suite and not be bothered with in all tests?

regards
/Pelle

**Comment by PelleUllberg** [comment]
I rather remove it from the HiveSQL annotation. It's still possible to explicitly set it in a specific test case by using the HiveRunnerSetup annotation.

```
@HiveRunnerSetup
    public final static HiveRunnerConfig CONFIG = new HiveRunnerConfig(){{
        setTimeoutEnabled(true);
        setTimeoutSeconds(30);
        setTimeoutRetries(2);
    }};
````

**Comment by PelleUllberg** [comment]
Looks good. Could you provide a comment in the Readme that explains how to configure the CompatibilityMode. Maybe we also should try to think of a better name than CompatibilityMode? HQLInterpreterMode, HQLParserMode or HQLCompatibilityMode?

**Comment by PelleUllberg** [comment]
Looks awesome. Thanks a bunch! I'll merge it.

/Pelle

**Pull request merged**